### PR TITLE
GH-7: Implement Construction Frame

### PR DESCRIPTION
**Pull request by Owen-Griffin**
* Issue: gh-7

### CODE REVIEW

File: backend/api.py [hunk 1]
```diff
@@ -87,3 +87,19 @@ def get_market_orders(self):
         response = self.request(route)
         orders = [MarketOrder.from_data(order) for order in response.json()]
         return orders
+
+    def get_buildings(self):
+        route = Route("/buildings", "GET")
+        response = self.request(route)
+        return response
```
Comment (ItsNeil17) [inline backend/api.py:94]: use response.json() here. won't work without

**Review by ItsNeil17 — CHANGES_REQUESTED** [review]
Upgrade buttons don't actually do anything I believe.

**Comment by Owen-Griffin** [comment]
I have made the requested changes; please review again. Be warned that app will not launch until API methods are implemented, as the methods are required in the `__init__` function of the frame.

**Review by ItsNeil17 — CHANGES_REQUESTED** [review]
Update frame levels when an upgrade has been done, like in alliance frame

**Comment by Owen-Griffin** [comment]
> Update frame levels when an upgrade has been done, like in alliance frame

This already happens:

```
self.parent.change_frame(
            ConstructionFrame(self.parent)
        )  # TODO instead of initialising a new frame, update the current frame
```

This is the same way it is done in the alliance frame.

**Review by ItsNeil17 — APPROVED** [review]
LGTM

**Pull request merged**